Tools/CISupport/ews-build/twisted_additions.py

```diff
@@ -219,7 +247,8 @@ def request(cls, url, type=None, params=None, headers=None, logger=None, timeout
             else:
                 agent = Agent(reactor, connectTimeout=timeout)
 
-            response = yield agent.request(typ, url.encode('utf-8'), Headers(headers))
+            body = JSONProducer(data) if data else None
```

Comment:
Do we assume data will never be an empty string or 0?
Also, is data to be None a valid input? json.dumps(None) works.

Reply:
We assume `data` is a dictionary that needs to be `json` encoded....probably just better to call it `json`, I suppose